src/backend/distributed/executor/multi_router_executor.c

```diff
@@ -410,12 +382,45 @@ RouterExecutorRun(QueryDesc *queryDesc, ScanDirection direction, long count)
 								   (int) operation)));
 		}
 
-		resultsOK = ExecuteTaskAndStoreResults(queryDesc, task,
-											   isModificationQuery,
-											   sendTuples);
-		if (!resultsOK)
+		if (requiresMasterEvaluation)
 		{
-			ereport(ERROR, (errmsg("could not receive query results")));
+			ListCell *taskCell = NULL;
+			Query *query = workerJob->jobQuery;
+			Oid relationId = ((RangeTblEntry *) linitial(query->rtable))->relid;
+
+			ExecuteMasterEvaluableFunctions(query);
```

Comment:
> Pass a Query * down for every Task instead of the Job, but that would incur a lot of overhead and requires a lot of refactoring on the physical planner side

What about doing that only for multi-task router planer? Would that be too poor in terms of the design? 
I think performance is not a very big concern for INSERT/SELECT case, since it has already a huge planning cost compared to the other modify queries and I don't expect we have a very strict planning performance requirements here. 

 
I've just [tried](https://gist.github.com/onderkalaci/72263afdb8280607142e3168ab7d50e0) that, and it worked for `now()`. We've a syncup with @anarazel and I'll also chat with him on this.

Btw, for the time-being, I plan to disallow all kinds of volatile functions.